msibi/pair.py

```diff
@@ -23,113 +20,72 @@ class Pair(object):
-        calculate_target_rdf : bool, optional, default=False
-            If True, the Freud python package (via the cmeutils package)
-            is used to calcualte the pair-wise RDF between pair.type1
-            and pair.type2 using state.traj_file. The RDF related
-            parameters are set in state.opt (msibi.optimize.MSIBI)
-        pair_indices : array-like (n_pairs, 2) dtype=int
-            Each row gives the indices of two atoms representing a pair
-            (default None)
-        alpha_form : str
-            For alpha as a function of r, gives form of alpha function
-            (default 'linear')
         """
-        if calculate_target_rdf and target_rdf != None:
-            raise ValueError(
-                    "Setting calcualte_target_rdf = True will overwirte "
-                    "the data passed to target_rdf. calculate_target_rdf "
-                    "should only be used when target_rdf is None"
-                    )
-        if target_rdf:
-            if os.path.isfile(target_rdf):
-                try:
-                    target_rdf = np.loadtxt(target_rdf)
-                except Exception as e:
-                    print(e)
-                    raise OSError("Unable to open target_rdf file "
-                            "Check that the path exists and is accessible."
-                            )
-            elif isinstance(target_rdf, np.ndarray):
-                pass
-
-        elif calculate_target_rdf:
-            target_rdf = state_pair_rdf(state, self)
-        else:
-            raise ValueError("At least one of target_rdf or "
-                    "calculate_target_rdf = True must be given."
-                    )
+        target_rdf = self.get_state_rdf(state)
 
-        if len(target_rdf) != state.opt.n_rdf_points:
-            raise ValueError(
-                    "The target RDF passed is not the same length as "
-                    "n_rdf_points set during the initialization of the "
-                    "MSIBI() class."
-                    )
-
-        self.states[state] = {
+        self._states[state] = {
             "target_rdf": target_rdf,
             "current_rdf": None,
             "alpha": state.alpha,
-            "alpha_form": alpha_form,
-            "pair_indices": pair_indices,
+            "alpha_form": "linear",
+            "pair_indices": None,
             "f_fit": [],
             "path": state.dir
         }
 
+    def get_state_rdf(self, state, exclude_bonded=True):
+        """Calculate the RDF of a Pair at a State."""
+        rdf, norm = gsd_rdf(
+                state.traj_file,
+                self.type1,
+                self.type2,
+                start=-state._opt.max_frames,
+                r_max=state._opt.rdf_cutoff,
+                bins=state._opt.n_rdf_points,
+                exclude_bonded=exclude_bonded
+                )
+        return np.stack((rdf.bin_centers, rdf.rdf*norm)).T
+
     def select_pairs(self, state, exclude_up_to=0):
```

Comment:
Do we need this function anymore now that we are using freud and bead type names for calcualte RDFs?

Reply:
select_pairs?